x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type_params.ts

```diff
@@ -19,15 +20,39 @@ export interface EsQueryAlertState extends AlertTypeState {
   latestTimestamp: string | undefined;
 }
 
-export const EsQueryAlertParamsSchemaProperties = {
-  index: schema.arrayOf(schema.string({ minLength: 1 }), { minSize: 1 }),
-  timeField: schema.string({ minLength: 1 }),
-  esQuery: schema.string({ minLength: 1 }),
+const EsQueryAlertParamsSchemaProperties = {
   size: schema.number({ min: 0, max: ES_QUERY_MAX_HITS_PER_EXECUTION }),
   timeWindowSize: schema.number({ min: 1 }),
   timeWindowUnit: schema.string({ validate: validateTimeWindowUnits }),
   threshold: schema.arrayOf(schema.number(), { minSize: 1, maxSize: 2 }),
-  thresholdComparator: schema.string({ validate: validateComparator }),
+  thresholdComparator: schema.string({ validate: validateComparator }) as Type<Comparator>,
+  searchType: schema.oneOf([schema.literal('esQuery'), schema.literal('searchSource')]),
```

Comment:
Do we need a migration to add this parameter to all existing Elasticsearch query rules?

Reply:
We assume `searchType` to be `esQuery` as a default, so it should work without migration. However I agree that we should add a migration in a follow up step, since we also need to take care of search source in persisted state migration.

Reply:
@ymao1 is this fine for you? added #127873 to track this

Reply:
I think it's fine to do as a followup as long as the migration makes the same release as this PR does!

Reply:
@kertal Left some minor comments in my review but overall looks good! Just need the migration :)

Reply:
thx for the feedback @ymao1 
@Dmitriynj is there a way to tweak the schema so `searchType` is for now optional (so internally we use the esQuery type by default). If not I think we should apply basic migration in this PR (without the searchSource migration)

Reply:
I have made `searchType` optional.

Reply:
Created migration PR #129179